src/subcommands/subcommand.go

```diff
@@ -84,27 +84,6 @@ func baseHelp() {
 	fmt.Println("\nUse 'semantifly <subcommand> --help' for more information about a specific subcommand.")
 }
 
-func setupSemantifly() {
-	semantifly_dir := flag.String("semantifly_dir", os.ExpandEnv("$HOME/.semantifly"), "Where to read existing semantifly data, and write new semantifly data.")
-	semantifly_index := flag.String("semantifly_index", "default", "By default, semantifly writes data to the 'default' subdir of the configured semantifly_dir. Setting this to a value other than 'default' allows for multiple indices on the same local machine.")
-
-	createDirectoryIfNotExists(*semantifly_dir)
-	index_path := path.Join(*semantifly_dir, *semantifly_index)
-	createDirectoryIfNotExists(index_path)
-
-	indexLog := path.Join(index_path, "index.log")
-	appendToIndexLog(indexLog)
-}
-
-func createDirectoryIfNotExists(dir string) {
```

Comment:
I don't understand why we can get rid of this, where else is this functionality provided?

Reply:
function is not used anywhere else, directories are only created & handled in add

Reply:
Yes, I'm asking you where we handle directory creation in add

Reply:
Since you missed this during testing, please modify the unit tests to use a random directory (eg /tmp/<UUID>) path so that we do not miss errors in this area in the future

Reply:
done

Reply:
oh the unit tests already used a random directory but the directories were being created before any subcommands were called, should i change that?

Reply:
Since we want the add subcommand to be able to create directories, we should remove explicit directory creation from the tests and have the subcommand create them instead

Reply:
hmm okay, i think it might be hard to make temp directories / bad practice to make temp directories for testing without os.MkDirTemp. Is it okay if just make a separate test for add just to test making the directories if they don't already exist?

Reply:
just do it in the $HOME directory then

Reply:
> Is it okay if just make a separate test for add just to test making the directories if they don't already exist?
Sure

Reply:
fixed